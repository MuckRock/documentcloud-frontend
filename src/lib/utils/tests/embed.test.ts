import { describe, test, expect } from "vitest";
import {
  settingsConfig,
  createEmbedSearchParams,
  getEmbedSettings,
  defaultSettings,
  isEmbed,
  reroute,
} from "../embed";

describe("embed settings", () => {
  test("settingsConfig", () => {
    expect(settingsConfig).toMatchSnapshot();
  });

  test("createEmbedSearchParams", () => {
    expect(createEmbedSearchParams({ title: null }).toString()).toEqual("");
  });

  test("getEmbedSettings", () => {
    const url = new URL("https://www.documentcloud.org");
    url.searchParams.set("pdf", "false");
    url.searchParams.set("onlyshoworg", "true");
    url.searchParams.set("title", "0");
    expect(getEmbedSettings(url.searchParams)).toEqual({
      ...defaultSettings,
      pdf: false,
      onlyshoworg: true,
      title: 0,
    });
  });
});

describe("embed utilities", () => {
  test("isEmbed", () => {
    const urls: [string, boolean][] = [
      [
        "https://www.documentcloud.org/documents/2622-agreement-between-conservatives-and-liberal-democrats-to-form-a-coalition-government/",
        false,
      ],
      [
        "https://www.documentcloud.org/documents/2622-agreement-between-conservatives-and-liberal-democrats-to-form-a-coalition-government/?embed=1",
        true,
      ],
      [
        "https://embed.documentcloud.org/documents/2622-agreement-between-conservatives-and-liberal-democrats-to-form-a-coalition-government/",
        true,
      ],
      [
        "https://embed.documentcloud.org/documents/2622-agreement-between-conservatives-and-liberal-democrats-to-form-a-coalition-government/?embed=1",
        true,
      ],
      [
        "https://www.documentcloud.org/documents/2622-agreement-between-conservatives-and-liberal-democrats-to-form-a-coalition-government/?embed",
        true,
      ],
    ];

    for (const [url, embed] of urls) {
      expect(isEmbed(new URL(url))).toEqual(embed);
    }
  });

  test("embed reroute", () => {
    // [original, rewritten]
    const paths: [string, string][] = [
      ["/embed/documents/1-slug/", "/embed/documents/1-slug/"],
      ["/documents/1-slug/?embed=1", "/embed/documents/1-slug/"],
      [
        "https://embed.documentcloud.org/documents/1-slug/",
        "/embed/documents/1-slug/",
      ],
    ];

    for (const [original, rewritten] of paths) {
      const url = new URL(original, "https://www.documentcloud.org");
      expect(reroute({ url })).toEqual(rewritten);
    }
  });
});
