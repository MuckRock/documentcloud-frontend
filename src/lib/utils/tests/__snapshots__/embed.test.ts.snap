// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`settingsConfig 1`] = `
{
  "embed": {
    "defaultValue": 1,
    "field": {
      "type": "hidden",
    },
    "storageIndex": null,
  },
  "fullscreen": {
    "defaultValue": null,
    "field": {
      "label": "dialogDocumentEmbedDialog.fullscreenOption",
      "options": [
        {
          "help": "dialogDocumentEmbedDialog.fsVisibleHelp",
          "label": "dialogDocumentEmbedDialog.visibleDefault",
          "value": null,
        },
        {
          "help": "dialogDocumentEmbedDialog.fsHiddenHelp",
          "label": "dialogDocumentEmbedDialog.hidden",
          "value": 0,
        },
      ],
      "type": "toggle",
    },
    "storageIndex": 5,
  },
  "height": {
    "defaultValue": null,
    "field": {
      "automatic": {
        "help": "dialogDocumentEmbedDialog.heightAuto",
        "label": "appearanceDimension.responsive",
        "value": null,
      },
      "fixed": {
        "help": "dialogDocumentEmbedDialog.widthFixed",
        "label": "appearanceDimension.fixed",
        "value": 500,
      },
      "label": "dialogDocumentEmbedDialog.height",
      "type": "dimension",
    },
    "storageIndex": 2,
  },
  "onlyshoworg": {
    "defaultValue": null,
    "field": {
      "label": "dialogDocumentEmbedDialog.contributedByFormat",
      "options": [
        {
          "help": "dialogDocumentEmbedDialog.cbfUserAndOrgHelp",
          "label": "dialogDocumentEmbedDialog.cbfUserAndOrg",
          "value": null,
        },
        {
          "help": "dialogDocumentEmbedDialog.cbfOrgOnlyHelp",
          "label": "dialogDocumentEmbedDialog.cbfOrgOnly",
          "value": 1,
        },
      ],
      "type": "toggle",
    },
    "storageIndex": 7,
  },
  "pdf": {
    "defaultValue": null,
    "field": {
      "label": "dialogDocumentEmbedDialog.pdfLink",
      "options": [
        {
          "help": "dialogDocumentEmbedDialog.plVisibleHelp",
          "label": "dialogDocumentEmbedDialog.visibleDefault",
          "value": null,
        },
        {
          "help": "dialogDocumentEmbedDialog.plHiddenHelp",
          "label": "dialogDocumentEmbedDialog.hidden",
          "value": 0,
        },
      ],
      "type": "toggle",
    },
    "storageIndex": 4,
  },
  "responsive": {
    "defaultValue": 1,
    "field": {
      "label": "dialogDocumentEmbedDialog.responsive",
      "options": [
        {
          "help": "dialogDocumentEmbedDialog.respOnHelp",
          "label": "dialogDocumentEmbedDialog.respOn",
          "value": 1,
        },
        {
          "help": "dialogDocumentEmbedDialog.respOffHelp",
          "label": "dialogDocumentEmbedDialog.respOff",
          "value": 0,
        },
      ],
      "type": "toggle",
    },
    "storageIndex": 8,
  },
  "sidebar": {
    "defaultValue": null,
    "field": {
      "label": "dialogDocumentEmbedDialog.sidebarBehavior",
      "options": [
        {
          "help": "dialogDocumentEmbedDialog.sbResponsiveHelp",
          "label": "dialogDocumentEmbedDialog.sbResponsive",
          "value": null,
        },
        {
          "help": "dialogDocumentEmbedDialog.sbHiddenHelp",
          "label": "dialogDocumentEmbedDialog.hidden",
          "value": 0,
        },
        {
          "help": "dialogDocumentEmbedDialog.sbVisibleHelp",
          "label": "dialogDocumentEmbedDialog.visible",
          "value": 1,
        },
      ],
      "type": "toggle",
    },
    "storageIndex": 0,
  },
  "text": {
    "defaultValue": null,
    "field": {
      "label": "dialogDocumentEmbedDialog.textMode",
      "options": [
        {
          "help": "dialogDocumentEmbedDialog.tmVisibleHelp",
          "label": "dialogDocumentEmbedDialog.visibleDefault",
          "value": null,
        },
        {
          "help": "dialogDocumentEmbedDialog.tmHiddenHelp",
          "label": "dialogDocumentEmbedDialog.hidden",
          "value": 0,
        },
      ],
      "type": "toggle",
    },
    "storageIndex": 6,
  },
  "title": {
    "defaultValue": 1,
    "field": {
      "label": "dialogDocumentEmbedDialog.titleBehavior",
      "options": [
        {
          "help": "dialogDocumentEmbedDialog.tVisibleHelp",
          "label": "dialogDocumentEmbedDialog.visibleDefault",
          "value": 1,
        },
        {
          "help": "dialogDocumentEmbedDialog.tHiddenHelp",
          "label": "dialogDocumentEmbedDialog.hidden",
          "value": 0,
        },
      ],
      "type": "toggle",
    },
    "storageIndex": 3,
  },
  "width": {
    "defaultValue": null,
    "field": {
      "automatic": {
        "help": "dialogDocumentEmbedDialog.widthAuto",
        "label": "appearanceDimension.responsive",
        "value": null,
      },
      "fixed": {
        "help": "dialogDocumentEmbedDialog.widthFixed",
        "label": "appearanceDimension.fixed",
        "value": 500,
      },
      "label": "dialogDocumentEmbedDialog.width",
      "type": "dimension",
    },
    "storageIndex": 1,
  },
}
`;
